src/libData/AccountData/Account.cpp

```diff
@@ -66,59 +66,27 @@ void Account::InitStorage() {
   }
 }
 
-bool Account::InitContract(const bytes& data, const Address& addr) {
-  SetInitData(data);
-  if (!InitContract(addr)) {
+bool Account::InitContract(const bytes& initData, const Address& addr) {
+  m_address = addr;
```

Comment:
I think originally this was placed at the end of the function to prevent prematurely setting `m_address` before all the checks pass, to avoid setting account to invalid state.  Any reason why we can't maintain the code?

Also, if you agree to put it back at the end, maybe we can also set `m_initData.clear();` if `InitContract` failed?
But anyway if the change was necessary, you can ignore this suggestion.

Reply:
Oh and we can use `CleanInitData` before returning false also?